### DISCUSSION TITLE
Database programming error 2.10.3

### DISCUSSION THREAD

**Discussion by hubert-mazur** [netbox-community/netbox#5700] [General]
### Environment
- Python version: 3.6.8
- NetBox version: 2.10.3
- PostgreSQL version: 12.3
- OS: Centos 7

### Steps to reproduce
- Create new database for NetBox
- Run migrations
- Perform other required actions from docs (configure nginx, collectstatic etc.)
- Enter NetBox UI

### Expected behavior
- No errors, app running smoothly

### Observed behavior

```
 A database programming error was detected while processing this request. Common causes include the following:

Database migrations missing - When upgrading to a new NetBox release, the upgrade script must be run to apply any new database migrations. You can run migrations manually by executing python3 manage.py migrate from the command line.

Unsupported PostgreSQL version - Ensure that PostgreSQL version 9.6 or higher is in use. You can check this by connecting to the database using NetBox's credentials and issuing a query for SELECT VERSION().

The complete exception is provided below:

<class 'django.db.utils.ProgrammingError'>

column dcim_interface._connected_interface_id does not exist
LINE 1: ...OUNT(*) AS "__count" FROM "dcim_interface" WHERE ("dcim_inte...


```

I tried to add this column manually, others are missing.

**Comment by candlerb** [comment]
* Is this an installation from scratch, or an upgrade?
* Have you modified or customised Netbox in any way at all?
* Did you see any errors at this step? https://netbox.readthedocs.io/en/stable/installation/3-netbox/#run-the-upgrade-script
* If an upgrade, did you do "git pull", or did you unzip over an existing installation? (The latter is wrong)
* If an upgrade, did you remember to restart? (`systemctl restart netbox netbox-rq`)

It looks to me like you're running the wrong version of Netbox for the database, since the dcim_interface table is not supposed to have a `_connected_interface_id` column anyway.  Here's what I see (with Netbox 2.10.4):

```
netbox=# \d dcim_interface
                                          Table "public.dcim_interface"
       Column        |          Type          | Collation | Nullable |                  Default
---------------------+------------------------+-----------+----------+--------------------------------------------
 id                  | integer                |           | not null | nextval('dcim_interface_id_seq'::regclass)
 name                | character varying(64)  |           | not null |
 type                | character varying(50)  |           | not null |
 mgmt_only           | boolean                |           | not null |
 description         | character varying(200) |           | not null |
 device_id           | integer                |           | not null |
 mac_address         | macaddr                |           |          |
 lag_id              | integer                |           |          |
 enabled             | boolean                |           | not null |
 mtu                 | integer                |           |          |
 mode                | character varying(50)  |           | not null |
 untagged_vlan_id    | integer                |           |          |
 cable_id            | integer                |           |          |
 _name               | character varying(100) |           | not null |
 label               | character varying(64)  |           | not null |
 _cable_peer_id      | integer                |           |          |
 _cable_peer_type_id | integer                |           |          |
 _path_id            | integer                |           |          |
Indexes:
...
```

Therefore, randomly adding columns is definitely not going to fix your problem.

If this was an upgrade, I suggest you:
1. Revert your postgres database to the last backup you took before the upgrade
2. Re-run the `./upgrade.sh`
3. Report any errors you see

**Reply by hubert-mazur** [reply]
This was installation from scratch.
Anyway, I downloaded newest release (2.10.4) and proceeded installation from docs. It works now like a charm. 
I think that the issue was running migrations manually - python3 manage.py migrate. This time, I run only an update script and works, no errors.

Thanks for Your time!
